app/src/main/java/com/example/tasky/MainActivity.kt

```diff
@@ -23,14 +30,61 @@ class MainActivity : ComponentActivity() {
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colorScheme.background
                 ) {
+                    ObserveError()
+                    ObserveLogin()
+                    ObserveSignUp()
                     LoginScreen(
-                        loginState = LoginState(),
-                        onLoginEvent = {}
+                        loginState = loginViewModel.state,
+                        onLoginEvent = loginViewModel::onEvent
                     )
                 }
             }
         }
     }
+
+    @Composable
+    private fun ObserveError() {
+        LaunchedEffect(key1 = loginViewModel.state.errorMessage) {
+            if (loginViewModel.state.errorMessage.isNotEmpty()) {
```

Comment:
I'm personally a friend of making a field nullable if you want to indicate it doesn't exist (e.g. there is no error), since that's exactly what nullables are used for